diffusion2d.cxx

```diff
@@ -106,7 +106,7 @@ void Diffusion2D::precon(BoutReal, BoutReal gamma, BoutReal) {
   // Neutral gas diffusion
   // Solve (1 - gamma*Dnn*Delp2)^{-1} 
   if(!inv) {
-    inv = Laplacian::create();
+    auto inv = Laplacian::create();
```

Comment:
Note that this creates a new Laplacian, that is deleted at the end of the scope (or at least no further used / referenced).

In this case, `inv` outside the scope is not set, and will give a segfault in line 117.